CONTRIBUTING.md

```diff
@@ -85,6 +89,75 @@ Because we will only merge a Pull Request for which all tests pass. The followin
   See [About TypeScript demos](#about-typescript-demos).
 - The Pull Request title follows the pattern `[Component] Imperative commit message`. (See: [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#imperative) for a great explanation)
 
+#### Checks and how to fix them
+
+If any of the checks fails click on the _Details_
+link and checkout the logs of the build to find out why it failed. The following
+section gives an overview what each check is responsible for.
+
+##### ci/codesandbox
+
+This task should not fail in isolation. It creates codesandboxes that use the version
+of Material-UI that was built from this Pull Request. Use it to test more complex scenarios.
+
+##### ci/circleci: checkout
+
+A preflight check to see if the dependencies and lockfile is ok. Running `yarn`
+and `yarn deduplicate` should fix most of the issues.
+
+##### ci/circleci: test_static
+
+Checks code format and lints the repository. The log of the failed build should explain
+how to fix the issues.
+
+##### ci/circleci: test_unit-1
+
+Runs the unit tests in a `jsdom` environment. If this fails then `yarn test:unit`
+should fail locally as well.
+
+##### ci/circleci: test_browser-1
+
+Runs the unit tests in multiple browsers (via Browserstack). The log of the failed
+build should list which browsers failed. If Chrome failed then `yarn test:karma`
+should fail locally as well. If other browsers failed debugging might be trickier.
+
+##### ci/circleci: test_regression-1
+
+Renders tests in `test/regressions/tests` and makes screenshots. This step shouldn't
+fail if the others pass. Otherwise a maintainer will take a look. The screenshots
+are evaluated in another step.
+
+##### argos
+
+Evaluates the screenshots taken in `test/regressions/tests` and fails if it detects
+differences. This doesn't necessarily mean your Pull Request will be rejected as a failure
+might be intended. Clicking on _Details_ will show you the differences.
+
+#### ci/circleci: test_types
+
+Typechecks the repository. The log of the failed build should list all issues.
+
+##### deploy/netlify
+
+Renders a preview of our docs with your changes if it succeeds. Otherwise `yarn docs:build`
+or `yarn docs:export` usually fail locally as well.
+
+##### mui-org.material-ui (Azure Pipelines)
+
+This task is mostly responsible for monitoring the bundle size. It will only report
+the size and not fail if it exceeds a certain threshold. If it fails there's usually
+something wrong with the how the packages or docs were built.
+
+#### codecov/project
+
+Monitors coverage of our tests. A reduction in coverage isn't necessarily bad but
+it's always appreciated if we can improve it.
```

Comment:
```suggestion
it's always appreciated if it can be improved.
```

Reply:
Right. There was a discussion a few weeks back about `we/our` usage.